process-invoices/invoice_steps.md

```diff
@@ -0,0 +1,164 @@
+
+4.  Create invoice data files (one for monographs and one for serials) to send to SAP.
+    Command:
+      `/usr/bin/python3.8 -m pipenv run python get_invoices_from_file.py [invoice_ids_YYYYMMDDhhmmss.txt filename from step 1]`
+    Outputs:
+      - File dlibsapg.###1.YYYYMMDD000000: Monographs SAP invoice data file (where ###1
+        is the next sequence number according to seq-sap)
+      - File dlibsapg.###2.YYYYMMDD000000: Serials SAP invoice data file (where ###2 is
+        the next sequence number according to seq-sap, which should one more than the
+        sequence number for the monographs SAP invoice file)
+      - File seq-sap gets updated by this command,  with the two sequence numbers used
+        for the SAP invoice files in this step appended. We are required by SAP to
+        submit sequentially numbered files and not skip or repeat any numbers. This
+        step ensures that future runs will increment each file number submitted to SAP
+        in the correct order.
+    Note:
+      If for some reason you have to rerun this "Final Run" process for the same set
+      of invoices, make sure to roll back the seq-sap file first.
+
+5.  Create monographs control file to send to SAP.
+    Command:
+      `/usr/bin/python3.8 -m pipenv run python make_cfile.py [dlibsapg.###1.YYYYMMDD000000 filename from step 4]`
+    Outputs:
+      - File clibsapg.###1.YYYYMMDD000000: Monographs SAP invoice control file
+    Note:
+      You can optionally check the control file by running `/usr/bin/python3.8 -m
+      pipenv run python parse_cfile.py [clibsapg.###1.YYYYMMDD000000 filename from this
+      step]`. This will print human-readable information from the control file to std-
+      out for verification.
+
+6.  Create serials control file to send to SAP.
+    Command:
+      `/usr/bin/python3.8 -m pipenv run python make_cfile.py [dlibsapg.###2.YYYYMMDD000000 filename from step 4]`
+    Outputs:
+      - File clibsapg.###2.YYYYMMDD000000: Serials SAP invoice control file
+    Note:
+      You can optionally check the control file by running `/usr/bin/python3.8 -m
+      pipenv run python parse_cfile.py [clibsapg.###2.YYYYMMDD000000 filename from this
+      step]`. This will print human-readable information from the control file to std-
+      out for verification.
+
+7.  Create monographs summary file.
+    Command:
+      `/usr/bin/python3.8 -m pipenv run python make_summary.py dlibsapg.###1.YYYYMMDD000000 > mono_summary_YYYYMMDD.txt`
+    Outputs:
+      - File mono_summary_YYYYMMDD.txt: Monographs summary file
+
+8.  Create serials summary file.
+    Command:
+      `/usr/bin/python3.8 -m pipenv run python make_summary.py dlibsapg.###2.YYYYMMDD000000 > ser_summary_YYYYMMDD.txt`
+    Outputs:
+      - File ser_summary_YYYYMMDD.txt: Serials summary file
+
+9.  Transmit invoice data and control files to SAP via FTP (soon to be sftp)
+    Commands, in order:
+      - `ftp sap-dropbox.mit.edu`
+      - `[username]`
```

Comment:
@zotoMIT should this be updated to reflect key-based access?

Reply:
Yes, I will update this with the new commands